packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -56,9 +56,10 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
 
   // TODO(pbo-linaro): Add support for windows-arm64 platform, https://github.com/flutter/flutter/issues/129807
   const TargetPlatform targetPlatform = TargetPlatform.windows_x64;
-  final Directory buildDirectory = globals.fs.directory(
-    getWindowsBuildDirectory(targetPlatform)
-  );
+  final Directory buildDirectory = globals.fs.directory(globals.fs.path.join(
+    projectPath,
+    getWindowsBuildDirectory(targetPlatform),
+  ));
```

Comment:
Question to self: why wasn't this a bug?

Reply:
If I might interrupt your conversation with yourself (since I was scratching my head hard over this one), `getWindowsBuildDirectory` is a relative path that relies on the fact that the current working directory MUST be a flutter app root in order to call this function. I violated this rule when I implemented `flutter build _preview`, and decided to make this explicit.